src/network/netlink.rs

```diff
@@ -405,3 +405,73 @@ pub fn parse_create_link_options(msg: &mut LinkMessage, options: CreateLinkOptio
         msg.nlas.push(Nla::NetNsFd(options.netns));
     }
 }
+
+#[cfg(test)]
+mod tests {
+    // Note this useful idiom: importing names from outer (for mod tests) scope.
+    use super::*;
+
+    macro_rules! test_setup {
+        () => {
+            if !nix::unistd::getuid().is_root() {
+                // there is no actual way to mark a test as skipped
+                // https://internals.rust-lang.org/t/pre-rfc-skippable-tests/14611
+                eprintln!("test skipped, requires root");
+                return;
+            }
+            nix::sched::unshare(nix::sched::CloneFlags::CLONE_NEWNET)
+                .expect("unshare(CLONE_NEWNET)");
+        };
+    }
+
+    macro_rules! run_command {
+        ($command:expr  $(, $args:expr)*) => {
+            std::process::Command::new($command).args([$($args),*]).output()
+                .expect("failed to run command")
+        };
+    }
+
+    #[test]
+    fn test_socket_new() {
+        test_setup!();
+        assert!(Socket::new().is_ok(), "Netlink Socket::new() should work");
+    }
+
+    #[test]
+    fn test_add_link() {
+        test_setup!();
+        let mut sock = Socket::new().expect("Socket::new()");
+
+        let name = String::from("test1");
+        sock.create_link(CreateLinkOptions::new(name.clone(), InfoKind::Dummy))
+            .expect("create link");
```

Comment:
Same here with adding a failed. Just for consistency with the `test_add_addr` test.